src/Cli/dotnet/commands/dotnet-test/VSTestArgumentConverter.cs

```diff
@@ -1,58 +1,87 @@
 ﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+
 namespace Microsoft.DotNet.Cli
 {
-    using System;
-    using System.Collections.Generic;
-    using System.Linq;
-    using static BlameArgs;
-
     /// <summary>
-    /// Converts the given arguments to vstest parsable arguments
+    /// Converts the given arguments to vstest parsable arguments.
     /// </summary>
+    /// <remarks>
+    /// This converter is only used when running <c>dotnet test</c> with a dll/exe,
+    /// not when called for project, solution, directory, or with no path.
+    /// </remarks>
     public class VSTestArgumentConverter
     {
-        private const string verbosityString = "--logger:console;verbosity=";
+        private const string VerbosityString = "--logger:console;verbosity=";
 
-        private readonly Dictionary<string, string> ArgumentMapping = new Dictionary<string, string>
-        {
-            ["-h"] = "--help",
-            ["-s"] = "--settings",
-            ["-t"] = "--listtests",
-            ["-a"] = "--platform",
-            ["-l"] = "--logger",
-            ["-f"] = "--framework",
-            ["-d"] = "--diag",
-            ["--filter"] = "--testcasefilter",
-            ["--list-tests"] = "--listtests",
-            ["--test-adapter-path"] = "--testadapterpath",
-            ["--results-directory"] = "--resultsdirectory",
-            ["--arch"] = "--platform"
-        };
-
-        private readonly Dictionary<string, string> VerbosityMapping = new Dictionary<string, string>
-        {
-            ["q"] = "quiet",
-            ["m"] = "minimal",
-            ["n"] = "normal",
-            ["d"] = "detailed",
-            ["diag"] = "diagnostic"
-        };
-
-        private readonly string[] IgnoredArguments = new string[]
-        {
-            "-c",
-            "--configuration",
-            "--runtime",
-            "-o",
-            "--output",
-            "--no-build",
-            "--no-restore",
-            "--interactive",
-            "--testSessionId",
-            "--artifacts-processing-mode"
-        };
+        private static readonly ImmutableDictionary<string, string> s_argumentMapping
+            = new Dictionary<string, string>()
+            {
+                ["-h"] = "--help",
+                ["-s"] = "--settings",
+                ["-t"] = "--listtests",
+                // .NET 7 breaking change, before we had ["-a"] = "--testadapterpath",
```

Comment:
Added a comment here so that we don't have to know or lookup for breaking changes that occurred in the code.